### PR TITLE
Stylo: Add grid-gap shorthand

### PR DESCRIPTION
**Pull request by methyl**
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

Should I add any tests?

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16081)

### CODE REVIEW

**Comment by bholley** [comment]
or @Wafflespeanut 

Thanks a ton for helping out with this stuff @methyl! This work is critical to helping us ship Servo in Firefox this year.

File: components/style/properties/shorthand/position.mako.rs [hunk 1]
```diff
@@ -127,3 +127,27 @@
...
     }
 </%helpers:shorthand>
+
+<%helpers:shorthand name="grid-gap" sub_properties="grid-row-gap grid-column-gap"
+                    spec="https://drafts.csswg.org/css-grid/#propdef-grid-gap">
+  use properties::longhands::{grid_row_gap, grid_column_gap};
+
+  pub fn parse_value(context: &ParserContext, input: &mut Parser) -> Result<Longhands, ()> {
+      let row_gap = grid_row_gap::parse(context, input)?;
+      let column_gap = grid_column_gap::parse(context, input).unwrap_or(row_gap.clone());
```
Comment (Manishearth) [inline components/style/properties/shorthand/position.mako.rs]: This should use `input.try()`; otherwise the cursor won't reset after a failed parse.

**Review by Manishearth — CHANGES_REQUESTED** [review]

File: components/style/properties/shorthand/position.mako.rs [hunk 2]
```diff
@@ -127,3 +127,27 @@
...
+      Ok(Longhands {
+        grid_row_gap: row_gap,
+        grid_column_gap: column_gap,
+      })
+  }
+
+  impl<'a> ToCss for LonghandsToSerialize<'a>  {
+      fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+          self.grid_row_gap.to_css(dest)?;
+          dest.write_str(" ")?;
```
Comment (Manishearth) [inline components/style/properties/shorthand/position.mako.rs]: i think you should skip these two steps if the column gap is the same. check what firefox does.

**Comment by methyl** [comment]
@Manishearth  I think build fails because gaps are declared for gecko, so it won't get registered as longhand. Not sure what to do in this case

**Comment by Manishearth** [comment]
put `product="gecko"` on the shorthand declaration

File: components/style/properties/shorthand/position.mako.rs (hunk 2 above)
Comment (methyl) [inline components/style/properties/shorthand/position.mako.rs]: I added the change, although I've been looking around and can't get any idea what's the usecase of ToCss so I can confirm with Firefox behavior.

**Comment by Wafflespeanut** [comment]
@bors-servo r-

Needs squash.

**Comment by Wafflespeanut** [comment]
Thanks!

@bors-servo r=Manishearth

**Pull request merged**